pkg/apis/cartographer/v1alpha1/workload_test.go

```diff
@@ -1001,6 +1001,134 @@ func TestWorkload_Merge(t *testing.T) {
 				Image: "ubuntu:bionic",
 			},
 		},
+	}, {
+		name: "image without source",
+		seed: &Workload{},
+		update: &Workload{
+			Spec: WorkloadSpec{
+				Image: "ubuntu:bionic",
+			},
+		},
+		want: &Workload{
+			Spec: WorkloadSpec{
+				Image: "ubuntu:bionic",
+			},
+		},
+	}, {
+		name: "image with subpath",
+		seed: &Workload{
+			Spec: WorkloadSpec{
+				Image: "alpine:latest",
+				Source: &Source{
+					Subpath: "/sys",
+				},
+			},
+		},
+		update: &Workload{
+			Spec: WorkloadSpec{
+				Image: "ubuntu:bionic",
+			},
+		},
+		want: &Workload{
+			Spec: WorkloadSpec{
+				Image: "ubuntu:bionic",
+			},
+		},
+	}, {
+		name: "image with sources nill",
+		seed: &Workload{
+			Spec: WorkloadSpec{
+				Image: "alpine:latest",
+			},
+		},
+		update: &Workload{
+			Spec: WorkloadSpec{
+				Image:  "ubuntu:bionic",
+				Source: &Source{Subpath: "my-subpath"},
+			},
+		},
+		want: &Workload{
+			Spec: WorkloadSpec{
+				Image: "ubuntu:bionic",
+			},
+		},
+	}, {
+		name: "image with sources source image and subpath",
```

Comment:
nit: can be renamed to something like "workload with multiple sources and subpath" or something like that

Reply:
Renamed